docs/configs/janusgraph-cfg.md

```diff
@@ -482,6 +478,39 @@ Configuration options for CQL executor service which is used to process deserial
 | storage.cql.executor-service.max-shutdown-wait-time | Max shutdown wait time in milliseconds for executor service threads to be finished during shutdown. After this time threads will be interrupted (signalled with interrupt) without any additional wait time. | Long | 60000 | LOCAL |
 
+### storage.cql.grouping
+Configuration options for controlling CQL queries grouping
+
+
+| Name | Description | Datatype | Default Value | Mutability |
+| ---- | ---- | ---- | ---- | ---- |
+| storage.cql.grouping.keys-allowed | If `true` this allows multiple partition keys to be grouped together into a single CQL query via `IN` operator based on the keys grouping strategy provided (usually grouping is done by same token-ranges or same replica sets, but may also involve shard ids for custom implementations).
+Notice, that any CQL query grouped with more than 1 key will require to return a row key for any column fetched.
+This option is useful when less amount of CQL queries is desired to be sent for read requests in expense of fetching more data (partition key per each fetched value).
+Notice, different storage backends may have different way of executing multi-partition `IN` queries (including, but not limited to how the checksum queries are sent for different consistency levels, processing node CPU usage, disk access pattern, etc.). Thus, a proper benchmarking is needed to determine if keys grouping is useful or not per case by case scenario.
+This option can be enabled only for storage backends which support `PER PARTITION LIMIT`. As such, this feature can't be used with Amazon Keyspaces because it doesn't support `PER PARTITION LIMIT`.
+If this option is `false` then each partition key will be executed in a separate asynchronous CQL query even when multiple keys from the same token range are queried.
+Notice, the default grouping strategy is not taking into account shards. Thus, this might be inefficient with ScyllaDB storage backend. ScyllaDB specific keys grouping strategy should be implemented after the resolution of the [ticket #232](https://github.com/scylladb/java-driver/issues/232). | Boolean | false | MASKABLE |
+| storage.cql.grouping.keys-class | Full class path of the keys grouping execution strategy. The class should implement `org.janusgraph.diskstorage.cql.strategy.GroupedExecutionStrategy` interface and have a public constructor with two arguments `org.janusgraph.diskstorage.configuration.Configuration` and `org.janusgraph.diskstorage.cql.CQLStoreManager`.
+Shortcuts available:
+- `tokenRangeAware` - groups partition keys which belong to the same token range. Notice, this strategy is not taking into account shards. Thus, this might be inefficient with ScyllaDB storage backend.
```

Comment:
ScyllaDB default load balancing strategy routes query using `RoutingKey` to a specific shard which is responsible to answer data for that shard. 
Each shard is served by a separate CPU. I.e. if your node has 8 CPUs than it means this node has 8 shards. 
ScyllaDB shard-aware load balancing strategy calculates ShardID from the provided routing key. If some of your partition keys provided in the query are not related to the specific shard which currently serves your query then part of your query will be routed to a separate shard (separate CPU). This sometimes might be less efficient, but it's hard to say how much less efficient it can be. 
Unfortunately, ShardInfo is currently private in the ScyllaDB Java Driver and it's complicated to figure our ShardId from PartitionKey + Node (without using Reflection API). When this information is open then we could group by Node + ShardId which should be more optimal than grouping by Node only.